cmd/speep/main.go

```diff
@@ -116,48 +141,42 @@ func HandleGet(getCmd *flag.FlagSet, all *bool, allShort *bool, name *string, na
 			return fmt.Errorf("failed to acquire the existing list: %w", err)
 		}
 		name := *name
+		fmt.Println("Category  Name  Shortcut key")
+		//TODO: How to remove new line betwewn the print above and the print below
 		for _, shortcut := range shortcuts {
-			if strings.Contains(shortcut.ShortcutKey, name) {
-				fmt.Println("Name \t Shortcut key")
-				fmt.Printf("%v \t %v \n", shortcut.Name, shortcut.ShortcutKey)
+			// TODO: Add category filter here too
+			if strings.Contains(shortcut.Name, name) {
+				fmt.Printf("%v\t %v\t %v\n", shortcut.Category, shortcut.Name, shortcut.ShortcutKey)
 			}
 		}
 	}
 	return nil
 }
 
-func ValidateNewShortcutKey(addCmd *flag.FlagSet, name *string, key *string) error {
-	if *name == "" || *key == "" {
-		addCmd.PrintDefaults()
-		return errors.New("name and the shortcut key are required to add a shortcut key")
+func ValidateNewShortcutKey(category *string, name *string, key *string) error {
+	if *category == "" || *name == "" || *key == "" {
+		return errors.New("category, name and shortcut key are required to register new shortcut key. Check \"speep help\" for the usage.")
 	}
 	return nil
 }
 
-func HandleAdd(addCmd *flag.FlagSet, name *string, nameShort *string, newShortcut *string, newShortcutShort *string) error {
-	addCmd.Parse(os.Args[2:])
-
-	if *name == "" {
-		*name = *nameShort
-	}
-	if *newShortcut == "" {
-		*newShortcut = *newShortcutShort
-	}
+func handleAdd(addCmd *flag.FlagSet, category *string, name *string, newShortcut *string) error {
 
-	err := ValidateNewShortcutKey(addCmd, name, newShortcut)
+	err := ValidateNewShortcutKey(category, name, newShortcut)
 	if err != nil {
-		return nil
+		return fmt.Errorf("%s", err)
```

Comment:
This is a bad idea since it removes any wrapped errors. You probably mean:
```suggestion
		return err
```

Better yet, wrap the error and add some more information.

Reply:
Fixed!